### ISSUE TITLE
could only angle loss  restrain the confidence mask ?

### ISSUE THREAD

**Issue by DavideHe** [matteo-rizzo/fc4-pytorch#9]
the loss defined for training only is  angle loss. there is no any other constraint for mask, this is efficient for confidence.
I run some raw image with cwp file pth. the masks always are light on bottom-left and top-left.

**Comment by matteo-rizzo** [comment]
Hi,

Thank you for your comment. However, I'm not really getting your point. Are you suggesting to introduce a different loss function? Are you experiencing problems with the current implementation?

**Comment by matteo-rizzo** [comment]
Hi,

Thank you for your comment. However, I'm not really getting your point. Are you suggesting to introduce a different loss function? Are you experiencing problems with the current implementation?

**Comment by DavideHe** [comment]
> Hi,
> 
> Thank you for your comment. However, I'm not really getting your point. Are you suggesting to introduce a different loss function? Are you experiencing problems with the current implementation?

Yes，In your code of the loss ,there is only angle loss. there is no any about the 4th channel (FC4) constraint.  if training as above (only angle loss),the 4th channel must learn by itself. I am not sure it will learn it.
AND I run your pretrained model of trained_models/fc4_cwp/*pth on other dataset,and I save the 4th channel as mask. I found the mask show regular bright spot (the bottom-left will be bright always). it seems that the 4th channel doesn't learn the real feature.
AND I  add the first derivative loss on the 4th channel, it is not very well.

**Comment by matteo-rizzo** [comment]
Thanks for your feedback. I can see the reason for your concern. However, my implementation aimed to be faithful to the original implementation in the paper, which did not apply specific regularizations to the confidence masks. Please feel free to branch the repo and suggest possible improvements, that would be much appreciated. Also, it seems weird that you found a constant confidence mask on another dataset. Which dataset are you using for your experiments?

**Comment by DavideHe** [comment]
> Thanks for your feedback. I can see the reason for your concern. However, my implementation aimed to be faithful to the original implementation in the paper, which did not apply specific regularizations to the confidence masks. Please feel free to branch the repo and suggest possible improvements, that would be much appreciated. Also, it seems weird that you found a constant confidence mask on another dataset. Which dataset are you using for your experiments?

I test on my own collection dataset on my camera with colorchecker. I will do more trials on some training dataset to answer my questions in my heart.If I get some useful  conclusion, I want to be able to exchange them with you.

**Comment by matteo-rizzo** [comment]
That would be great, thank you!
